src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php

```diff
@@ -139,6 +139,25 @@ public function compileUnique(Blueprint $blueprint, Fluent $command)
         );
     }
 
+    /**
+     * Compile a unique index ignoring a null column command.
+     *
+     * @param  \Illuminate\Database\Schema\Blueprint  $blueprint
+     * @param  \Illuminate\Support\Fluent  $command
+     * @return string
+     */
+    public function compileUniqueIgnoreNull(Blueprint $blueprint, Fluent $command)
+    {
+        $columns = $command->columns;
+        $columns[] = $command->nullColumn;
```

Comment:
Just adding the column to the index works as SQL Server treats NULL values as the same. This is slightly less performant as a partial index as the null column is included in the index taking up storage / memory.